test/unistd/io.c

```diff
@@ -26,10 +26,8 @@ int main() {
 
     var device = FS.makedev(major++, 0);
     FS.registerDevice(device, {
-      open: function(stream) {
-        stream.payload = [65, 66, 67, 68];
-      },
       read: function(stream, buffer, offset, length, pos) {
+        stream.payload = [65, 66, 67, 68];
```

Comment:
Is it acceptable to modify existing tests? It looks like the use of `stream.payload` isn't used anywhere else in the code. Attaching a payload to stream in `open` makes compatibility difficult since JS_IMPL doesn't support `open`. This test ensures that WasmFS and the legacy API can use the exact same device operations.

Reply:
This seems like a reasonable change. But I think we should not reset the payload in each read. Perhaps
```suggestion
        if (!stream.payload) {
          stream.payload = [65, 66, 67, 68];
        }
```